src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -256,13 +256,12 @@ ss::future<ss::stop_iteration> async_manifest_view_cursor::next_iter() {
                                       : ss::stop_iteration::no;
 }
 
-std::optional<std::reference_wrapper<const partition_manifest>>
+ssx::task_local_ptr<const partition_manifest>
```

Comment:
>  It's not safe to store this value in
the temporary variable and dereference it after the scheduling point
because the underlying manifest may be altered.

I'm a little concerned about this. Why would we invent task local ptr to solve this rather than working to fit into the existing seastar model? For example, is this a unique situation that other techniques can't be used to solve?

cc @andrwng

Reply:
The original idea was to use an access method that takes a lambda and calls it with the manifest. The lambda prevents the user from using co_await. But this interface is not always convenient so I added another access method. I can remove task_local stuff if you feel strongly about it. I don't think it's used a lot. The code mostly uses method that takes a functor.